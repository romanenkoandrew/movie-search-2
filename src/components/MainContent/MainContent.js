import React from 'react';
import TableContainer from 'components/TableContainer';
import PropTypes from 'prop-types';
import { useLocation } from 'react-router-dom';
import { Button, Form, Input, Typography, Row, Col, Checkbox } from 'antd';
import styles from './MainContent.styles';

const { Text } = Typography;

const MainContent = (props) => {
  const { getAllTitles, currentPageReset } = props;
  const { pathname } = useLocation();
  const [searchBlockValues, setSearchBlockValues] = React.useState({
    search: '',
    type: '',
  });
  const onFinishSearchForm = (values) => {
    let type;
    const { search, movies, series } = values;
    if (movies) type = 'movie';
    else if (series) type = 'series';
    else type = '';
    currentPageReset();
    setSearchBlockValues({ search: search, type: type });
    getAllTitles({ search, type });
  };

  return (
    <>
      {' '}
      {pathname === '/' && (
        <Row justify='center'>
          <Col span={15}>
            <Form
              css={styles.form}
              layout='horizontal'
              name='searchForm'
              initialValues={{
                movies: false,
                series: false,
              }}
              onFinish={onFinishSearchForm}
            >
              <Form.Item
                name='search'
                rules={[
                  {
                    required: true,
                    message: 'Please input title of the movie!',
                  },
                ]}
              >
                <Input
                  placeholder='Enter the title of the movie'
                  autoComplete='off'
                />
              </Form.Item>

              <Row justify='space-between'>
                <Form.Item dependencies={['series']}>
                  {({ getFieldValue }) => {
                    return getFieldValue('series') ? (
                      <Form.Item name='movies' valuePropName='checked'>
                        <Checkbox disabled>Movies</Checkbox>
                      </Form.Item>
                    ) : (
                      <Form.Item name='movies' valuePropName='checked'>
                        <Checkbox>Movies</Checkbox>
                      </Form.Item>
                    );
                  }}
                </Form.Item>

                <Form.Item dependencies={['movies']}>
                  {({ getFieldValue }) => {
                    return getFieldValue('movies') ? (
                      <Form.Item name='series' valuePropName='checked'>
                        <Checkbox disabled>Series</Checkbox>
                      </Form.Item>
                    ) : (
                      <Form.Item name='series' valuePropName='checked'>
                        <Checkbox>Series</Checkbox>
                      </Form.Item>
                    );
                  }}
                </Form.Item>
                <Form.Item>
                  <Button type='primary' htmlType='submit'>
                    Search
                  </Button>
                </Form.Item>
              </Row>
            </Form>
          </Col>
        </Row>
      )}
      <TableContainer searchBlockValues={searchBlockValues} />
    </>
  );
};

MainContent.propTypes = {
  getAllTitles: PropTypes.func.isRequired,
  currentPageReset: PropTypes.func.isRequired,
};

export default MainContent;
