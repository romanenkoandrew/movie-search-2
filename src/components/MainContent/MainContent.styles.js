import { css } from '@emotion/core';

export default {
  form: () => css`
    margin-top: 1rem;
    .ant-form-item {
      margin-bottom: 0.5rem;
    }
  `,
};
