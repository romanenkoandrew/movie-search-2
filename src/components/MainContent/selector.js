import { sideMenuIsOpenSelector } from 'selectors';
import { createStructuredSelector } from 'reselect';

export default createStructuredSelector({
  sideMenuIsOpen: sideMenuIsOpenSelector,
});
