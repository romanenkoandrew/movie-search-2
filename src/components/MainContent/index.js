import MainContent from './MainContent';
import { connect } from 'react-redux';
import selector from './selector';
import { getAllTitles } from 'actions';

const mapDispatchToProps = { getAllTitles };

export default connect(selector, mapDispatchToProps)(MainContent);
