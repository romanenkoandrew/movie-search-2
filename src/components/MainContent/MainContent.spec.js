import React from 'react';
import { shallow } from 'enzyme';
import MainContent from './MainContent';

const initialState = {
  showAlert: () => {},
  currentPageReset: () => {},
  getAllTitles: () => {},
};
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: 'localhost:3000/',
  }),
}));
const setUp = () => shallow(<MainContent {...initialState} />);

describe('MainContent component', () => {
  let component;
  let instance;
  beforeEach(() => {
    component = setUp();
    instance = component.instance();
  });
  it('should render MainContent component', () => {
    expect(component).toMatchSnapshot();
  });
});
