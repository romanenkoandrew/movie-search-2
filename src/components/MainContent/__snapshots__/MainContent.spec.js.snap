// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContent component should render MainContent component 1`] = `
<Fragment>
  <Connect(TableContainer)
    searchBlockValues={
      Object {
        "search": "",
        "type": "",
      }
    }
  />
</Fragment>
`;
